### PR TITLE
Offering other url handlers on SecurityException

### LINKED ISSUE #6001: Crash report 7.4: SecurityException in ActivityLauncher.openUrlExternal
This is possibly related to https://github.com/wordpress-mobile/WordPress-Android/issues/4991 with a similar stack trace.  The crash is exclusive to Samsung devices running Android 7 (90%) and 6 (10%).

```
Fatal Exception: java.lang.SecurityException: Permission Denial: starting Intent { act=android.intent.action.VIEW dat=https://bitcoin770.wordpress.com/... cmp=com.alibaba.intl.android.apps.poseidon/com.alibaba.android.intl.weex.activity.WeexPageActivity launchParam=MultiScreenLaunchParams { mDisplayId=0 mFlags=0 } } from ProcessRecord{1f8a09b 24402:org.wordpress.android/u0a33} (pid=24402, uid=10033) not exported from uid 10258
       at android.os.Parcel.readException(Parcel.java:1693)
       at android.os.Parcel.readException(Parcel.java:1646)
       at android.app.ActivityManagerProxy.startActivity(ActivityManagerNative.java:3530)
       at android.app.Instrumentation.execStartActivity(Instrumentation.java:1526)
       at android.app.Activity.startActivityForResult(Activity.java:4399)
       at android.support.v4.app.BaseFragmentActivityJB.startActivityForResult(BaseFragmentActivityJB.java:50)
       at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:79)
       at android.app.Activity.startActivityForResult(Activity.java:4358)
       at android.support.v4.app.FragmentActivity.startActivityForResult(FragmentActivity.java:859)
       at android.app.Activity.startActivity(Activity.java:4682)
       at android.app.Activity.startActivity(Activity.java:4650)
       at org.wordpress.android.ui.ActivityLauncher.openUrlExternal(ActivityLauncher.java:350)
       at org.wordpress.android.ui.themes.ThemeWebActivity.openTheme(ThemeWebActivity.java:58)
       at org.wordpress.android.ui.themes.ThemeBrowserActivity.startWebActivity(ThemeBrowserActivity.java:455)
       at org.wordpress.android.ui.themes.ThemeBrowserActivity.onTryAndCustomizeSelected(ThemeBrowserActivity.java:472)
       at org.wordpress.android.ui.themes.ThemeBrowserFragment$2.onClick(ThemeBrowserFragment.java:224)
       at android.view.View.performClick(View.java:6199)
       at android.view.View$PerformClick.run(View.java:23647)
       at android.os.Handler.handleCallback(Handler.java:751)
       at android.os.Handler.dispatchMessage(Handler.java:95)
       at android.os.Looper.loop(Looper.java:154)
       at android.app.ActivityThread.main(ActivityThread.java:6682)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
```

### PR DESCRIPTION
**Pull request by hypest**
Fixes #6001 

The Alibaba B2B still causes problems (probably older versions of the app still in circulation) so, let's guard against the SecurityException and offer the user the list of available URL handlers (web browsers usually) to chose from.

Unfortunately, there are no steps to reproduce here. A way to test this is to manually modify the code to throw a SecurityException right after the call to `WPActivityUtils.disableComponent()`.

To test:
* no steps available

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 1]
```diff
@@ -374,18 +376,31 @@ public static void showAztecEditorReleaseNotes(Activity activity) {
...
+            AppLog.e(AppLog.T.UTILS, "Error opening url in default browser. Url: " + url, se);
+
+            List<ResolveInfo> infos = context.getPackageManager().queryIntentActivities(intent, 0);
+            if (infos.size() == 1) {
+                // there's only one handler and apparently it caused the exception so, just inform and bail
+                AppLog.d(AppLog.T.UTILS, "Only one url handler found so, bailing.");
+                ToastUtils.showToast(context, context.getString(R.string.cant_open_url));
+            } else {
+                Intent chooser = Intent.createChooser(intent, context.getString(R.string.error_please_choose_browser));
+                context.startActivity(chooser);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Niptick: Should we re-set the AppLock extended timeout here? `AppLockManager.getInstance().setExtendedTimeout();`

Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Hah, not a nitpick at all, good catch! Fixed with 16e9370.

**Comment by hypest** [comment]
Thanks for picking this up for review @daniloercoli , ready for another pass. Thanks!

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 2]
```diff
@@ -374,18 +376,32 @@ public static void showAztecEditorReleaseNotes(Activity activity) {
...
             ToastUtils.showToast(context, context.getString(R.string.no_default_app_available_to_open_link), ToastUtils.Duration.LONG);
             AppLog.e(AppLog.T.UTILS, "No default app available on the device to open the link: " + url, e);
+        } catch (SecurityException se) {
+            AppLog.e(AppLog.T.UTILS, "Error opening url in default browser. Url: " + url, se);
+
+            List<ResolveInfo> infos = context.getPackageManager().queryIntentActivities(intent, 0);
+            if (infos.size() == 1) {
+                // there's only one handler and apparently it caused the exception so, just inform and bail
+                AppLog.d(AppLog.T.UTILS, "Only one url handler found so, bailing.");
+                ToastUtils.showToast(context, context.getString(R.string.cant_open_url));
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Sorry If I didn't bring this up in the 1st review - can't we re-use the `R.string.no_default_app_available_to_open_link` error message here? Both seem very close in wording `Unable to open the link` and `Error occurred while opening link`.

Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: No problem, makes sense! Consolidated the two string resources with 886e431. I renamed the older one as it no longer corresponded to its usage. Ready for another pass, thanks!

**Comment by daniloercoli** [comment]
Looks good, ship it! :shipit:

**Pull request merged**